wat/WatLexer.g4

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright (c) 2019 Renata Hodovan.
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+   list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors
+   may be used to endorse or promote products derived from this software
+   without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+lexer grammar WatLexer;
+
+LPAR : '(' ;
+RPAR : ')' ;
+
+NAT : Nat ;
+INT : Int ;
+FLOAT : Float ;
+STRING : String ;
+VALUE_TYPE : NXX ;
+CONST : NXX '.const' ;
+
+FUNCREF: 'funcref' ;
+MUT: 'mut' ;
+
+NOP: 'nop' ;
+UNREACHABLE: 'unreachable' ;
+DROP: 'drop' ;
+BLOCK: 'block' ;
+LOOP: 'loop' ;
+END: 'end' ;
+BR: 'br' ;
+BR_IF: 'br_if' ;
+BR_TABLE: 'br_table' ;
+RETURN: 'return' ;
+IF: 'if' ;
+THEN: 'then' ;
+ELSE: 'else' ;
+SELECT: 'select' ;
+CALL: 'call' ;
+CALL_INDIRECT: 'call_indirect' ;
+
+LOCAL_GET: 'local.get' ;
+LOCAL_SET: 'local.set' ;
+LOCAL_TEE: 'local.tee' ;
+GLOBAL_GET: 'global.get' ;
+GLOBAL_SET: 'global.set' ;
+
+LOAD : NXX '.load' (MEM_SIZE '_' SIGN)? ;
+STORE : NXX '.store' (MEM_SIZE)? ;
+
+OFFSET_EQ_NAT : 'offset=' Nat ;
+ALIGN_EQ_NAT : 'align=' Nat ;
+
+UNARY
+  : IXX '.clz'
```

Comment:
Maybe it makes sense to introduce a token type for every instruction? It removes the need to check string values during parse tree handling (because tokens will be available).

The final result looks like this:

```antlr
unary
    : IxxClz
    | ...
    ;

IxxClz : IXX '.clz';
```